.github/index.pkl

```diff
@@ -0,0 +1,229 @@
+import "@pkl.impl.ghactions/helpers.pkl"
+
+local buildArtifactsName = "pkl-lsp-build-artifacts"
+
+testReports {
+  junit {
+    "build/test-results/**/*.xml"
+  }
+  excludeJobs {
+    Regex("deploy-.*")
+  }
+}
+
+local buildAndTest: Workflow = new {
+  jobs {
+    ["build"] = buildJob(false)
+  }
+}
+
+local function buildJob(isRelease: Boolean): Workflow.Job = new {
+  `runs-on` = "ubuntu-latest"
+  steps {
+    new Common.Checkout {
+      with {
+        `fetch-depth` = 0 // needed for spotless
+      }
+    }
+    new Setup.Java {
+      with {
+        `java-version` = "21"
+        distribution = "temurin"
+        cache = "gradle"
+      }
+    }
+    new {
+      `if` = "github.repository != 'apple/pkl-lsp'"
+      run =
+        """
+        git remote add upstream https://github.com/apple/pkl-lsp.git
+        git fetch upstream main
+        echo 'PKL_ORIGINAL_REMOTE_NAME=upstream' > "$GITHUB_ENV"
+        """
+    }
+    new {
+      local args: Listing<String> = new {
+        "--info"
+        "--stacktrace"
+        when (isRelease) {
+          "-DreleaseBuild=true"
+        }
+      }
+      run =
+        """
+        ./gradlew \(args.join(" ")) check shadowJar verifyDistribution
+        """
+    }
+    new Artifact.Upload {
+      name = "Upload build artifacts"
+      `if` = "!cancelled()"
+      with {
+        name = buildArtifactsName
+        path = "build/libs/"
+      }
+    }
+  }
+}
+
+local function verifyJobs(requiresJob: String): Mapping<String, Workflow.Job> = new {
+  default {
+    needs { requiresJob }
+    steps {
+      new Common.Checkout {}
+      new Setup.Java {
+        with {
+          `java-version` = "21"
+          distribution = "temurin"
+          cache = "gradle"
+        }
+      }
+      new Artifact.Download {
+        with {
+          name = buildArtifactsName
+          path = "build/libs/"
+        }
+      }
+      new {
+        run =
+          """
+          ./gradlew --info --stacktrace verifyDistribution
+          """
+      }
+    }
+  }
+  ["test-macos"] {
+    `runs-on` = new Listing { "self-hosted"; "macos" }
+    `if` = "github.repository_owner == 'apple'"
+  }
+  ["test-windows"] {
+    `runs-on` = "windows-latest"
+  }
+}
+
+local verifyJobNames = verifyJobs("build").keys.toListing()
+
+local baseMavenDeploy: Workflow.Job = new {
+  `runs-on` = "ubuntu-latest"
+  environment = "maven-release"
+  steps {
+    new Common.Checkout {}
+    new Setup.Java {
+      with {
+        `java-version` = "21"
+        distribution = "temurin"
+        cache = "gradle"
+      }
+    }
+    new Artifact.Download {
+      with {
+        name = buildArtifactsName
+        path = "build/libs/"
+      }
+    }
+  }
+}
+
+prb {
+  jobs {
+    ["build"] = buildJob(false)
+  }
+}
+
+build {
+  jobs {
+    ["build"] = buildJob(false)
+    ...(verifyJobs("build")) {
+      // use lower priority runners for on-push macOS jobs
+      ["test-macos"] { `runs-on` { "nightly" } }
```

Comment:
I don't think this works right now. Can we disable this and leave a TODO to re-enable?

CC @madrob